types/chart.js/index.d.ts

```diff
@@ -270,7 +271,7 @@ declare namespace Chart {
         fontColor?: ChartColor;
         fontFamily?: string;
         padding?: number;
-        generateLabels?(chart: any): any;
+        generateLabels?(chart: ChartData): ChartLegendLabelItem[];
```

Comment:
According to the documentation, it should be generateLabels?(chart: Chart): ChartLegendItem[];

Reply:
Thanks, I changed it to `Chart`

Reply:
Also ChartLegendLabelItem[] it's supposed to be ChartLegendItem[].

Reply:
No, it has the `datasetIndex` property and therefore is a `ChartLegendLabelItem`

Reply:
In that snippet they are attaching such property but the element is defined without it: https://www.chartjs.org/docs/latest/configuration/legend.html#legend-item-interface.

Actually, the link you shared is clearly identifying datasetIndex as an extra parameter.

Reply:
Then the definition of `onClick` must be updated too to reflect that change.

Reply:
@adripanico would you rather see the `onClick` method definition changed, or keep the `ChartLegendLabelItem`?

Reply:
It's ok. I'll approve it.